src/SprykerEco/Zed/Payone/Business/Distributor/OrderPriceDistributor.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Payone\Business\Distributor;
+
+use Generated\Shared\Transfer\OrderTransfer;
+use Generated\Shared\Transfer\PaymentTransfer;
+use SprykerEco\Zed\Payone\PayoneConfig;
+
+class OrderPriceDistributor implements OrderPriceDistributorInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\OrderTransfer
+     */
+    public function distributeOrderPrice(OrderTransfer $orderTransfer): OrderTransfer
+    {
+        if ($orderTransfer->getPayments()->count() <= 1) {
+            return $orderTransfer;
+        }
+
+        $payonePayment = $this->findPayonePayment($orderTransfer);
+        if (!$payonePayment) {
+            return $orderTransfer;
+        }
+
+        $totals = $orderTransfer->getTotals();
+
+        $roundingError = 0;
+        $priceRatio = $this->calculatePriceRatio($payonePayment->getAmount(), $totals->getGrandTotal());
+
+        $this->distributeOrderItemsPrices($orderTransfer, $priceRatio, $roundingError);
+        $this->distributeOrderExpensesPrices($orderTransfer, $priceRatio, $roundingError);
+
+        $orderTransfer->setTotals($totals->setGrandTotal($payonePayment->getAmount()));
+
+        return $orderTransfer;
+    }
+
+    /**
+     * @param int $paymentAmount
+     * @param int $grandTotalAmount
+     *
+     * @return float
+     */
+    protected function calculatePriceRatio(int $paymentAmount, int $grandTotalAmount): float
```

Comment:
Maybe we could use our `Decimal` class here?

Reply:
I guess that was skipped due to a new composer dependency. Also this solution is temporary and will be rewritten, so I 'd like to keep float here.